### ISSUE TITLE
Implement Elchemy REPL

### ISSUE THREAD

**Issue by wende** [wende/elchemy#154]
Functionality: 
A REPL should take single lines of Elchemy, and produce an output of compiled Elixir code:

Examples:
```
elchemy> 2 + 5
7
elchemy> List.map ((+) 2) [1,2,3]
[3,4,5]
elchemy> Just 10
{10}
elchemy> Nothing
nil
```

Ideas:
- Autocompletion on Tab press (Can be Elixir autocompletion since the syntax of variables and Module.function is the same)
- Display compiled Elixir code when demanded
- Compile the output back to Elchemy syntax (?)
- Display the type of the output (?)

Possible implementation:
- Node.js process wrapper around iex. Using Elchemy functions to compile String in and feed them out to wrapped process.
- Listen for tab key and either send tab key to the process or use `IEx.Autocomplete` module

**Comment by wende** [comment]
For output clarity probably #93 is a must

**Comment by wende** [comment]
Blocked by #164

**Comment by wende** [comment]
Not worth the effort. Closing in favor of #207

**Comment by wende** [comment]
Resurrecting for discussion on possible implementations

**Comment by wende** [comment]
I think there's an option to hack this pretty easily:

1. Write a simple `Repl.elm` file in Elchemy that understands the input to the repl and translates into related Elixir code
  a) If it looks like statement add to a module and reload the whole thing
  b) If it looks like expression treat it as a query and only print the output
2. Download the whole Elchemy codebase 
3. Write an `expect` script to feed input information into `elm repl` launched in Elchemy codebase directory, and pass it further into Repl.input() function.
4. If there were no errors take the output of code produced and feed it to `iex repl`
5. Print out the result of both repl's (iex result and elm-repl type result) to the interface
6. ???
7. Profit

**Comment by wende** [comment]
Would be way easier to ship if #352 was done
